scripts/parser-tests/typescript/index.js

```diff
@@ -1,37 +1,58 @@
 import path from "path";
-import fs from "fs/promises";
+import fs from "fs";
 import { fileURLToPath } from "url";
-import ts from "../../../build/typescript/lib/typescript.js";
 import TestRunner from "../utils/parser-test-runner.js";
-import parsingErrorCodes from "./error-codes.js";
+import ErrorCodes from "./error-codes.js";
+
+import { spawnSync } from "child_process";
+const getEncoding = path =>
+  spawnSync("file", ["--brief", "--mime-encoding", path])
```

Comment:
The typescript parser test is 5x slower (3s vs 16s) due to the process spawning overhead. Since there are only 7 tests failing if we assume they are `utf-8`:
```
     bom-utf16be.ts false true
     bom-utf16le.ts false true
     collisionCodeGenModuleWithUnicodeNames.ts false true
     instanceofOperator.ts false true
     promiseTest.ts false true
     targetTypeBaseCalls.ts false true
     unicodeIdentifierNames.ts false true
```
Can we read the test file as buffer and check the BOM instead? Or we can just hardcode those utf16 files.

Reply:
Let me see if I can read them all in one go (and thus only spawn once), would prefer to not have to mess around with BOMs.

Reply:
OK just ended up doing the BOM thing like you suggested. I tried looking at other PRs, and I see them mostly taking in the 9s range, while these now take 6s -- so it seems faster than without this PR (certainly faster than when I was shelling out to `file` of course), but just wanting to point out the actual times since you mentioned 3s, but perhaps that was just on your machine?

Reply:
Here is the commit, do un-resolve the conversation if you think it is still an issue: https://github.com/babel/babel/pull/14320/commits/f38dc5d93278c7fab9dbb7cf846bc40c4ff12652